File: account/src/main/java/serp/project/account/infrastructure/store/mapper/BaseMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.account.infrastructure.store.mapper;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

public abstract class BaseMapper {
    
    protected Long localDateTimeToLong(LocalDateTime localDateTime) {
        if (localDateTime == null) {
            return null;
        }
        return localDateTime.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli();
    }

    protected LocalDateTime longToLocalDateTime(Long timestamp) {
        if (timestamp == null) {
            return null;
        }
        return LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.systemDefault());
    }
}

File: ptm_task/src/infrastructure/store/mapper/base_mapper.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project - Base mapper utilities for Go services
*/

package mapper

import "time"

func TimeToEpochMilli(t time.Time) int64 {
	if t.IsZero() {
		return 0
	}
	return t.UnixMilli()
}

func EpochMilliToTime(epochMilli int64) time.Time {
	if epochMilli == 0 {
		return time.Time{}
	}
	return time.UnixMilli(epochMilli)
}

func GetCurrentEpochMilli() int64 {
	return time.Now().UnixMilli()
}
